woocommerce/woocommerce-ios
# [Mobile Payments] Printing receipts. Mark II: Yosemite

Closes #3979 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader ⚠️

We continue building our prototype (for reference, p91TBi-4q4-p2), this time, by adding a new store to manage the business logic related to receipts.


https://user-images.githubusercontent.com/2722505/115587100-453a2080-a29b-11eb-82ea-1458dff2f4c7.mov



In this first iteration, all we do is print a receipt, but soon (e.g. #3975) we will need to add support for persisting the data necessary to generate receipts for any eligible order.

Tradeoffs:
* The action to generate and print the receipt is triggered automatically after a payment succeeds. This is temporary, and it is not the final, expected behaviour. But it helps test this PR.

This PR starts where #4002 ends:

## Changes
* Add a new Store (ReceiptStore) and a new Action (ReceiptAction) to encapsulate the business logic of loading pre-existing receipts from Storage, and coordinate printing receipts to the ReceiptPrinterService introduced in PR #4002 
* Rename the model object that was passed as parameter to he ReceiptPrinterService. This is motivated because...
* We abandon the pre-exisitng idea of persisting PaymentIntent. Instead, we will persist only the information we need to generate a receipt. This is something that might have to be rolled back later. We'll see.
* Some unit tests.

## How to test
Testing is a bit complicated because:

1. Requires setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. Requires an external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* At that point, and only in order to test this PR, a preview of the receipt should be presented. At this point, the preview only shows a page number and the actual amount and currency of the order that was just charged using the card reader. The amount and currency should match the Order's.

Note: this is not the actual flow that will be implemented in the final beta. There is work pending in tickets  #3982, #3981 #3978

Update release notes:
